### ISSUE TITLE
Error: Expected all tensors to be on the same device, but found at least two devices

### ISSUE THREAD

**Issue by linthy94** [ssbuild/baichuan_finetuning#3]
when i set train_info_args['devices'] = [4,5,6,7], the program will print the error.
Error: 
……
Expected all tensors to be on the same device, but found at least two devices, cuda:5 and cuda:1! (when checking argument for argument index in method wrapper_CUDA__index_select)
……

But  i set train_info_args['devices'] = 4, the training program will succeed.

**Comment by ssbuild** [comment]
yeah ,set CUDA_VISIBLE_DEVICES=4,5,6,7 , and   train_info_args['devices'] = 4  also.
